src/Text/Pandoc/Writers/XWiki.hs

```diff
@@ -0,0 +1,256 @@
+inlineToXWiki (SmallCaps lst) = do
+  contents <- inlineListToXWiki lst
+  return contents
+
+inlineToXWiki (Quoted SingleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "'" <> contents <> "'"
+
+inlineToXWiki (Quoted DoubleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "\"" <> contents <> "\""
+
+inlineToXWiki (Code (_,classes,_) contents') = do
+  let at  = Set.fromList classes `Set.intersection` highlightingLangs
+  let contents = pack contents'
+  return $
+    case Set.toList at of
+      [] -> "{{code}}" <> contents <> "{{/code}}"
+      (l:_) -> "{{code language=\"" <> (pack l) <> "\"}}" <> contents <> "{{/code}}"
+
+inlineToXWiki (Cite _ lst) = inlineListToXWiki lst
+
+-- FIXME: optionally support this (plugin?) 
+inlineToXWiki (Math _ str) = return $ "{{formula}}" <> (pack str) <> "{{/formula}}"
+
+inlineToXWiki il@(RawInline frmt str)
+  | frmt == Format "xwiki" = return $ pack str
+  | otherwise              = "" <$ report (InlineNotRendered il)
+
+-- TODO: Handle anchors
+inlineToXWiki (Link _ txt (src, _)) = do
+  label <- inlineListToXWiki txt
+  case txt of
+     [Str s] | isURI src && escapeURI s == src -> return $ pack src
+     _  -> return $ "[[" <> label <> ">>" <> (pack src) <> "]]"
+
+inlineToXWiki (Image _ alt (source, tit)) = do
+  alt' <- inlineListToXWiki alt
+  let
+    titText = pack tit
+    params = intercalate " " [
+        if Text.null alt' then "" else "alt=\"" <> alt' <> "\"",
+          if Text.null titText then "" else "title=\"" <> titText <> "\""
+        ]
+  return $ "[[image:" <> (pack source) <> (if Text.null params then "" else "|| " <> params) <> "]]"
+
+inlineToXWiki (Note contents) = blockListToXWiki contents
+
+-- FIXME: support attrs
+inlineToXWiki (Span _ contents) = inlineListToXWiki contents
+  
+-- Utility method since (for now) all lists are handled the same way
+blockToXWikiList :: PandocMonad m => Text -> [[Block]] -> XWikiReader m Text
+blockToXWikiList marker contents = do
+  lev <- asks listLevel
+  contents' <- local (\s -> s { listLevel = listLevel s <> marker } ) $ mapM listItemToXWiki contents
+  return $ vcat contents' <> if Text.null lev then "\n" else ""
+  
+
+listItemToXWiki :: PandocMonad m => [Block] -> XWikiReader m Text
+listItemToXWiki contents = do
+  marker <- asks listLevel
+  contents' <- blockListToXWiki contents
+  return $ marker <> ". " <> contents'
+
+
+-- | Convert definition list item (label, list of blocks) to MediaWiki.
+definitionListItemToMediaWiki :: PandocMonad m
+                              => ([Inline],[[Block]])
+                              -> XWikiReader m Text
+definitionListItemToMediaWiki (label, items) = do
+  labelText <- inlineListToXWiki label
+  contents <- mapM blockListToXWiki items
+  marker <- asks listLevel
+  return $ marker <> " " <> labelText <> "\n" <>
+    intercalate "\n" (map (\d -> (Data.Text.init marker) <> ": " <> d) contents)
+
+escapeXWikiString :: Text -> Text
```

Comment:
Is this really all that needs escaping?
```
What if I write a dash--like this--and I don't want emphasis?
```

Reply:
Are you saying I should escape things like "--", "**", and "//"?

Reply:
I think I have this fixed, but there aren't any examples like this in testsuite.native

Reply:
Had to make a minor fix for this.

Tested with simple HTML input:

```html
<p>This--and not this--is ok.</p>
<p>So **is this**?</p>
<p>And what //about this//?</p>
```

produces:

```xwiki
This~-~-and not this~-~-is ok.

So ~*~*is this~*~*?

And what ~/~/about this~/~/?
```